operators_evaluation.go

```diff
@@ -0,0 +1,33 @@
+package repository
+
+import "github.com/globalsign/mgo/bson"
+
+func Regex(field string, value interface{}, options string) BinaryOperator {
+	operator := "$regex"
+	if options == "" {
+		options = "i"
+	}
+	return &BinaryOperatorImpl{
+		Type:      BinaryOperatorTypeRegex,
+		OpField:   &operator,
+		FieldName: field,
+		Value: bson.RegEx{
+			Pattern: value.(string),
+			Options: options,
+		},
+	}
+}
+
+type FindText struct {
+	Search             string  `bson:"$search"`
+	Language           *string `bson:"$language,omitempty"`
+	CaseSensitive      bool    `bson:"$caseSensitive,omitempty"`
+	DiacriticSensitive bool    `bson:"$diacriticSensitive,omitempty"`
+}
+
+func Text(value ... interface{}) *BooleanOperator {
```

Comment:
The only option for $text operator is to pass your `FindText` structure. So ask for it in the params. Do not ask for a "value...interface{}" ask for a *FindText pointer.

Reply:
Hum, I understand. I Changed the argument type to `Find Text` in function.